File: python/exercises/inheritance/FibonacciProgression.py
from Progression import Progression

class FibonacciProgression(Progression):
    """Iterator producing Fibonacci series."""

    def __init__(self, first=0, second=1):
        """Create a new fibonacci progression.

        first   the first term of the progression (default 0)
        second  the second term of the progression (default 1)
        """
        super().__init__(first)
        self._prev = second - first

    def _advance(self):
        """ Replace (prev, current) with (current, current + prev)"""
        self._prev, self._current = self._current, self._current + self._prev

File: java/exercises/inheritance/FibonacciProgression.java
/** Generate the fibonacci progression */
public class FibonacciProgression extends Progression {

  // instance variable
  protected long prev;

  /** Contruct traditional Fibonacci, 0, 1, 1, 2, 3, 5...*/
  public FibonacciProgression() { this(0, 1); }

  /** Construct generalized Fibonacci give first and second values */
  public FibonacciProgression(long first, long second) {
    super(first);
    // ficticious preceding value for first
    prev = second - first;
  }

  /** Replace (prev, current) with (current, current+prev) */
  protected void advance() {
    long temp = prev;
    prev = current;
    current += temp;
  }
}
